### PR TITLE
fix: use color and variant for AListItem

### PR DESCRIPTION
**Pull request by ManUtopiK**
This change add the possibility to use layerProps (color and variant) to `AListItem` with default to `text`.
If `color` is provided with `variant`, `is-active` item use this variant and hovered item use `light` variant.

```vue
...
<AListItem color="danger" title="Delete" />
...
```
image

```vue
// docs/demos/list/DemoListVModelSupport.vue#L39
...
<AListItem
    v-for="(item, index) in items"
    color="info"
    variant="fill"
...
```
image

### TODO :
- [ ] update doc

### CODE REVIEW

**Comment by ManUtopiK** [comment]
Is this change acceptable ? I write the doc ?

**Comment by jd-solanki** [comment]
> Is this change acceptable ? I write the doc ?

I dill review it soon

File: packages/anu-vue/src/components/list-item/AListItem.tsx [hunk 1]
```diff
@@ -81,10 +81,10 @@ export const AListItem = defineComponent({
 
     // useLayer
     const { styles, classes } = getLayerClasses(
-      computed(() => props.isActive ? props.color || 'primary' : undefined),
-      computed(() => props.isActive ? props.variant || 'light' : 'text'),
+      computed(() => props.color ?? props.isActive ? props.color || 'primary' : undefined),
+      computed(() => props.isActive ? (!props.variant && !props.color ? 'text' : props.variant || 'light') : 'text'),
       toRef(props, 'states'),
-      { statesClass: 'states:10' },
+      computed(() => ({ statesClass: props.isActive ? 'states:10' : 'hover:states:10' })),
```
Comment (jd-solanki) [inline packages/anu-vue/src/components/list-item/AListItem.tsx:87]: @ManUtopiK What does this line do?

Comment (jd-solanki) [reply packages/anu-vue/src/components/list-item/AListItem.tsx:87]: I mean what's the need for `hover:states:10'`?

Comment (ManUtopiK) [reply packages/anu-vue/src/components/list-item/AListItem.tsx:87]: I don't remember exactly why it was for. But I struggled a little bit with states before #86. 

Also, there are still some improvements to do.

1. We lose the state of the `isActive` item when using together a `color` AND the variant `text`.

Basic example without color and text:

image

With props `color="success" variant="text"`:

image
`Donut jujubes` isn't highlighted.

2. We can add color when there is no click handler.
```vue
<AListItem
  title="Send"
  color="success"
/>
```
`color` props do nothing here.

**Comment by ManUtopiK** [comment]
Arff, I'm really sorry for the big mess I made on this branch. I wanted to update and use #86 but I made a mistake.
I don't understand the state here on gitub. [My branch](https://github.com/ManUtopiK/anu/tree/AListItemLayerProps) seems good ; 3 commits ahead `jd-solanki:main`.
Git is hard :/

It's late. I will close this pull request and create a new one tomorrow...
Sorry again for the noise.

**Comment by ManUtopiK** [comment]
I close this pull request for now. I broke the branch anyway.
I have to think on component `state` and `variant` to go further.

**Comment by jd-solanki** [comment]
Hi @ManUtopiK 

I am refactoring the whole codebase from TSX to SFC so if want to create a new PR, please wait for 1 or 2 days to get latest updates.

**Comment by ManUtopiK** [comment]
> I am refactoring the whole codebase from TSX to SFC

Woah, a lot of changes ! I've worked on a loading component. I didn't finish. I'm back after Christmas.

**Comment by jd-solanki** [comment]
Hi @ManUtopiK refactoring PR is merged! 😇

**Pull request closed without merging**